monai/metrics/surface_distance.py

```diff
@@ -0,0 +1,179 @@
+    We require that images are the same size, and assume that they occupy the
+    same space (spacing, orientation, etc.).
+
+    Args:
+        seg_1: first binary or labelfield image.
+        seg_2: second binary or labelfield image.
+        label_idx: for labelfield images, convert to binary with
+            `seg_1 = seg_1 == label_idx`.
+        crop: crop input images and only keep the foregrounds. In order to
+            maintain two inputs' shapes, here the bounding box is achieved
+            by ``(seg_1 | seg_2)`` which represents the union set of two
+            images. Defaults to ``True``.
+    """
+
+    # Get both labelfields as np arrays
+    if torch.is_tensor(seg_1):
+        seg_1 = seg_1.detach().cpu().numpy()
+    if torch.is_tensor(seg_2):
+        seg_2 = seg_2.detach().cpu().numpy()
+
+    # Check non-zero number of elements and same shape
+    if seg_1.size == 0 or seg_1.shape != seg_2.shape:
+        raise ValueError("Labelfields should have same shape (and non-zero number of elements)")
+
+    # If not binary images, convert them
+    if seg_1.dtype != bool:
+        seg_1 = seg_1 == label_idx
+    if seg_2.dtype != bool:
+        seg_2 = seg_2 == label_idx
+
+    # Check both have at least 1 voxel with desired index
+    if not (np.any(seg_1) and np.any(seg_2)):
+        raise ValueError(f"Labelfields should have at least 1 voxel containing the desired labelfield, {label_idx}")
+
+    if crop:
+        seg_1, seg_2 = np.expand_dims(seg_1, 0), np.expand_dims(seg_2, 0)
+        box_start, box_end = generate_spatial_bounding_box(seg_1 | seg_2)
+        cropper = SpatialCrop(roi_start=box_start, roi_end=box_end)
+        seg_1, seg_2 = np.squeeze(cropper(seg_1)), np.squeeze(cropper(seg_2))
+
+    # Do binary erosion and use XOR to get edges
+    edges_1 = binary_erosion(seg_1) ^ seg_1
+    edges_2 = binary_erosion(seg_2) ^ seg_2
+
+    return (edges_1, edges_2)
+
+
+def get_surface_distance(
+    seg_1: Union[np.ndarray, torch.Tensor],
+    seg_2: Union[np.ndarray, torch.Tensor],
+    label_idx: int,
+    crop: bool = True,
+) -> np.ndarray:
+    """
+    This function is used to compute the surface distances from `seg_1` to `seg_2`.
+
+    In order to improve the computing efficiency, before getting the edges,
+    the images can be cropped and only keep the foreground if not specifies
+    ``crop = False``.
+
+    Args:
+        seg_1: first binary or labelfield image.
+        seg_2: second binary or labelfield image.
+        label_idx: for labelfield images, convert to binary with
+            `seg_1 = seg_1 == label_idx`.
+        crop: crop input images and only keep the foregrounds. In order to
+            maintain two inputs' shapes, here the bounding box is achieved
+            by ``(seg_1 | seg_2)`` which represents the union set of two
+            images. Defaults to ``True``.
+    """
+    (edges_1, edges_2) = get_mask_edges(seg_1, seg_2, label_idx, crop)
+    dis_1 = distance_transform_edt(~edges_2)
```

Comment:
`distance_transform_edt` has the option of `metric : {‘chessboard’, ‘taxicab’}, optional` please make it available in `get_surface_distance `

Reply:
Hi @wyli , do you mean [.distance_transform_cdt](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.distance_transform_cdt.html#scipy.ndimage.distance_transform_cdt) ? Thus, we should use distance_transform_edt (Exact Euclidean distance) or  distance_transform_cdt (chessboard or taxicab) ?

Reply:
I see, I thought there is a unified interface for these in ndimage. I don't think chessboard/taxicab is widely used. but we could have `compute_percent_hausdorff_distance ` with `metric` and `percentile` as parameters, this generic logic is useful